zerver/views/auth.py

```diff
@@ -236,7 +250,29 @@ def finish_google_oauth2(request):
     else:
         logging.error('Google oauth2 account email not found: %s' % (body,))
         return HttpResponse(status=400)
+
     email_address = email['value']
+    request.authed_subdomain = subdomain  # Used by middleware to set cookie
+
+    subdomain_login_uri = ''.join([
+        settings.EXTERNAL_URI_SCHEME,
+        subdomain,
+        '.',
+        settings.EXTERNAL_HOST,
+        reverse('zerver.views.auth.log_into_subdomain')
+    ])
+
+    response = redirect(subdomain_login_uri)
```

Comment:
For some reason this is still not working for me. I have added the commits which do exactly what you are saying.